### PR TITLE
owner(cdc): backport stale metrics fix to 5.3

### PR DESCRIPTION
**Pull request by liuzix**
### What problem does this PR solve?
Issue Number: ref #4774

### What is changed and how it works?
- This PR is the combined cherry-pick of both #4775 and #4998

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test

### Release note 

```release-note
Fix TiCDC incorrectly display stale metrics data on dashboard.
```

### CODE REVIEW

**Review by asddongmen — APPROVED** [review]

File: cdc/owner/owner.go [hunk 1]
```diff
@@ -25,15 +25,16 @@ import (
 	"github.com/pingcap/errors"
 	"github.com/pingcap/failpoint"
 	"github.com/pingcap/log"
+	pd "github.com/tikv/pd/client"
+	"go.uber.org/zap"
+	"golang.org/x/time/rate"
+
```
Comment (hi-rustin) [inline cdc/owner/owner.go:31]:
```suggestion
```

**Review by hi-rustin — APPROVED** [review]
rest LGTM

**Pull request merged**